### PR TITLE
Feature/simd load

### PR DESCRIPTION
**Pull request by rrahn**
Adds load function for simd.

Tested with sse4, avx2 and avx512.

### CODE REVIEW

**Comment by rrahn** [comment]
@marehr can u review this?

File: include/seqan3/core/simd/simd_algorithm.hpp [hunk 1]
```diff
@@ -77,6 +78,28 @@ constexpr simd_t iota(typename simd_traits<simd_t>::scalar_type const offset)
...
+    if constexpr (simd_traits<simd_t>::max_length == 16)
+        return reinterpret_cast<simd_t>(_mm_loadu_si128(reinterpret_cast<__m128i const *>(mem_addr)));
+    else if constexpr (simd_traits<simd_t>::max_length == 32)
+        return reinterpret_cast<simd_t>(_mm256_loadu_si256(reinterpret_cast<__m256i const *>(mem_addr)));
+    else if constexpr (simd_traits<simd_t>::max_length == 64)
+        return reinterpret_cast<simd_t>(_mm512_loadu_si512(mem_addr));
+    else if constexpr (simd_traits<simd_t>::max_length == 1)  // scalar value?
+        return simd_t{*reinterpret_cast<simd_t const *>(mem_addr)};
+    else
+        static_assert(simd_traits<simd_t>::max_length >= 1 && simd_traits<simd_t>::max_length,
```
Comment (marehr) [inline include/seqan3/core/simd/simd_algorithm.hpp]: `simd_traits<simd_t>::max_length` is weird.

File: test/unit/core/simd/simd_algorithm_test.cpp [hunk 2]
```diff
@@ -38,3 +38,20 @@ TEST(simd_algorithm, iota)
     constexpr simd_type result = iota<simd_type>(0);
     SIMD_EQ(result, expect);
 }
+
+TEST(simd_algorithm, load)
+{
+    using simd_type = simd_type_t<int16_t>;
+
+    std::array<int16_t, simd_traits<simd_type>::length> mem{};
+    simd_type expect{};
```
Comment (marehr) [inline test/unit/core/simd/simd_algorithm_test.cpp]:
```suggestion
    simd_type expect = iota<simd_type>(0);
```

File: test/unit/core/simd/simd_algorithm_test.cpp [hunk 3]
```diff
@@ -38,3 +38,20 @@ TEST(simd_algorithm, iota)
...
+
+    for (int16_t i = 0; i < static_cast<int16_t>(mem.size()); ++i)
+    {
+        mem[i] = i;
+        expect[i] = i;
```
Comment (marehr) [inline test/unit/core/simd/simd_algorithm_test.cpp]:
```suggestion

```

**Comment by rrahn** [comment]
superseded by #1190

**Pull request closed without merging**